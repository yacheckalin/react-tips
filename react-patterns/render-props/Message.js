import React from "react";

const Card = (props) => (
  <div className="row">
    <div className="col s12 m6">
      <div className="card blue-grey darken-1">
        <div className="card-content white-text">
          <span className="card-title">Card Title</span>
          <p>{props.children()}</p>
        </div>
      </div>
    </div>
  </div>
);

const Message = () => (
  <div>
    <Card>
      {() => (
        <>
          I am a very simple card. I am good at containing small bits of
          information. I am convenient because I require little markup to use
          effectively.
        </>
      )}
    </Card>
  </div>
);

export default Message;
