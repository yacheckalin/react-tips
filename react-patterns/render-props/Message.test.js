import React from "react";
import toJSON from "enzyme-to-json";
import Message from "./Message";
import { shallow } from "enzyme";

describe("<Message />", () => {
  it("renders and displays properly", () => {
    const wrapper = shallow(<Message />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
