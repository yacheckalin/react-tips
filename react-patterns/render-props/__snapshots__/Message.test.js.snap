// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Message /> renders and displays properly 1`] = `
<div>
  <Card>
    <Component />
  </Card>
</div>
`;
